lightning/src/onion_message/messenger.rs

```diff
@@ -304,6 +314,101 @@ where
 	}))
 }
 
+/// Decode one layer of an incoming onion message
+/// Returns either a Forward (another onion message), or Receive (decrypted content)
+pub fn peel_onion<NS: Deref, L: Deref, CMH: Deref, T: CustomOnionMessageContents>(
```

Comment:
Unused type bound here:
```suggestion
pub fn peel_onion<NS: Deref, L: Deref, CMH: Deref>(
```